packages/codemods/src/codeMods/__tests__/ComponentToCompat/ComponentToCompat.test.ts

```diff
@@ -0,0 +1,84 @@
+import { Project } from 'ts-morph';
+import * as Root from '../mock/compat/mockIndex';
+import {
+  IRawCompat,
+  runComponentToCompat,
+  getNamedExports,
+  buildCompatHash as buildHash,
+  repathNamedImports,
+  repathPathedImports,
+} from '../../mods/ComponentToCompat/CompatHelpers';
+import * as Button from 'office-ui-fabric-react/lib-commonjs/Button';
+
+const createComponentToCompat = (compat: IRawCompat) => {
+  return {
+    oldPath: './mockIndex',
+    newComponentPath: 'compat/Button',
+    namedExports: getNamedExports(compat.namedExports),
+  };
+};
+
+describe('Component to compat', () => {
+  let project: Project;
+  beforeEach(() => {
+    project = new Project();
+    project.addSourceFilesAtPaths(`${process.cwd()}/**/__tests__/mock/compat/*.tsx`);
+  });
+
+  it('correctly repaths from exact known import', () => {
+    const file = project.getSourceFileOrThrow('ImportsStuff.tsx');
+    const hash = buildHash([{ componentName: 'Button', namedExports: Root }], createComponentToCompat);
+    repathPathedImports(file, hash);
+    expect(file.getFullText()).toContain('compat/Button');
+  });
+
+  it('correctly gets maps imports from an external library', () => {
+    const buttonExports = getNamedExports(Button);
+    expect(buttonExports).toContain('DefaultButton');
+    expect(buttonExports).toContain('CommandButton');
+  });
+
+  it('correctly repaths from index', () => {
+    const file = project.getSourceFileOrThrow('ImportsStuff.tsx');
```

Comment:
I'm having trouble understanding what some of the rest of these do too, since you're using fake files which don't actually reference OUFR.

Reply:
It's making sure that the utilities/each individual piece does the job that it's supposed to do. That way when they all get put together it will work as expected.

Reply:
Part of what I meant by that was that it would be helpful to have more comments explaining what each one does. And I'm still having trouble how it works with fake files which don't actually reference OUFR.